flink-yarn/src/main/java/org/apache/flink/yarn/cli/FlinkYarnSessionCli.java

```diff
@@ -319,13 +306,7 @@ public AbstractYarnClusterDescriptor createDescriptor(String defaultApplicationN
 			yarnClusterDescriptor.setTaskManagerSlots(slots);
 		}
 
-		String[] dynamicProperties = null;
-		if (cmd.hasOption(DYNAMIC_PROPERTIES.getOpt())) {
-			dynamicProperties = cmd.getOptionValues(DYNAMIC_PROPERTIES.getOpt());
-		}
-		String dynamicPropertiesEncoded = StringUtils.join(dynamicProperties, YARN_DYNAMIC_PROPERTIES_SEPARATOR);
-
-		yarnClusterDescriptor.setDynamicPropertiesEncoded(dynamicPropertiesEncoded);
+		GlobalConfiguration.setDynamicProperties(retrieveDynamicProperties(cmd, DYNAMIC_PROPERTIES));
```

Comment:
Can we get around setting a static field in `GlobalConfiguration`? Why not parsing the dynamic properties and then passing them to the `ClusterDescriptor` at creation time. I guess it would also make sense to do the configuration loading outside of the `ClusterDescriptor`.